### ISSUE TITLE
No sounds or voices

### ISSUE THREAD

**Issue by tangsgod** [organicmaps/organicmaps#584]
I don't have sounds and voices..
I can't turn on the sound icon neither the voice option in parameters.

Lineage OS 17.1 Android 10 Redmi Note 8
Current version: 2021.06.07

**Comment by biodranik** [comment]
Do you have a TTS engine installed? There are some instructions here: https://mapsme.zendesk.com/hc/en-us/articles/208628985-How-can-I-check-TTS-settings-on-my-Android-device-

**Comment by tangsgod** [comment]
Thank you @bernikovich , the engine wasn't intalled. I did it then I can tick the sounds option.
While i can't test it now at work, i'll try it later on the road back home.

**Comment by dominikstrb** [comment]
I'll just post to this issue because I have a similar problem. I have TTS installed (https://f-droid.org/packages/com.github.olga_yakovleva.rhvoice.android) and it is working in the system settings but not in Organic Maps. I can enable and disable TTS in the app's settings, but when I start a navigation, it stays silent.

**Comment by biodranik** [comment]
@dominikstrb please share `adb logcat` logs after enabling TTS in the app's settings and restarting the app.

**Comment by dominikstrb** [comment]
Hi @biodranik, here is my [log](https://github.com/organicmaps/organicmaps/files/6847861/log.txt). Thanks! If there is anything else I can do, please let me know.

**Comment by biodranik** [comment]
What is your device's locale? I see that TTS languages are not detected by our code for some reason. Either we didn't query it properly, or the package is missing something for proper integration.

**Comment by dominikstrb** [comment]
My language is set to English (US).

**Comment by biodranik** [comment]
Does it work in some other TTS apps, excluding system settings?

**Comment by dominikstrb** [comment]
Yes, it works for example in Osmand.

**Comment by biodranik** [comment]
@rtsisyk Looks like the current code has some bug that doesn't allow to detect TTS properly on Android side. Can you please take a look? Maybe it's locale-related.

**Comment by Endem10n** [comment]
@dominikstrb could you, please also clarify Android version and device model?

**Comment by dominikstrb** [comment]
I'm using Android 11 (CalyxOS 2.7.1) on a Google Pixel 4a.

**Comment by biodranik** [comment]
Looks like there is a bug: if system TTS settings were changed, OM doesn't get these changes without a restart. So please relaunch the app after any TTS system modifications.

**Comment by PetbkA** [comment]
Same here:
**OS**: LineageOS 19 (Android 12)
**OM**: 2022.03.23-4-Google
**TTS Engines:**
1. [Google](https://play.google.com/store/apps/details?id=com.google.android.tts&hl=en&gl=US) googletts.google-speech-apk_20220321.02_p1.436777630
2. [RHVoice](https://f-droid.org/packages/com.github.olga_yakovleva.rhvoice.android/) 1.8.0

Behavior is:
1. If system language is set to English, no TTS Engines are [seen](https://user-images.githubusercontent.com/22843816/165888805-528def32-97b7-438f-ade9-d2309181310c.jpg)
in OM, although they [are seen](https://user-images.githubusercontent.com/22843816/165888859-8a946c31-da3d-4157-ba26-5b1fe2ae4e37.jpg) in OS Settings;
2. Trying to enable voice instructions in this state only change switch to On, but only text changes, [not the switch state](https://user-images.githubusercontent.com/22843816/165888978-46b79590-b4fd-449e-aefc-303eb9c5f350.jpg);
3. If I'll change OS language to Russian, I [can enable](https://user-images.githubusercontent.com/22843816/165889036-82207dbb-4bf5-46c6-af01-4a0f3ac73ad6.jpg) voice instructions in OM;
4. Luckily changing OS language back to English doesn't disable instructions in OM, so step 3 is a workaround.

Here's the [logs](https://github.com/organicmaps/organicmaps/files/8588289/logs.zip) for 1 and 2.

**Comment by biodranik** [comment]
I think the problem is related to the outdated code of the TTS engine API, and it doesn't work properly on modern devices.
It would be great if someone can check it.

**Comment by biodranik** [comment]
There is a [workaround](https://github.com/RHVoice/RHVoice/issues/510#issuecomment-1073370897) for RHVoice to work properly: to install ESpeak, and then restart the app.

**Comment by foss-** [comment]
Just another data point: LineageOS 18.1 on gauguin device: no voice output with both Google Speech Services and RHVoice installed.

**Comment by biodranik** [comment]
@foss- did you try to install ESpeak and reboot the device?

**Comment by foss-** [comment]
installed eSpeak (looks deprecated, there is an 8 year old version and a 1 year old version, installed the latter). Rebooted device. Selected eSpeak in `System Settings > Accessibility > Text-to-speech output > Preferred engine` but no dice. Should I hear voice instruction when starting a navigation or do I have to be moving? Asking because testing this from a desk. eSpeak sounds rather horrible when playing a test snippet in system settings.

Sidenote: `OM > Settings > Voice Instructions > For more information please check this guide` takes me to a maps.me support article. Not sure if a KB exists for OM but that should probably be adjusted.

**Comment by biodranik** [comment]
@foss- That is a known issue, there is a work going on: https://github.com/organicmaps/organicmaps.github.io/pull/32

Espeak [version](https://f-droid.org/packages/com.reecedunn.espeak/) on FDroid is built in 2021. 

Did you try to switch to RHVoice after installing ESpeak?

**Comment by foss-** [comment]
No, so tried now:
- switch system to `RHVoice` as Preferred engine. When then starting a navigation, still hearing nothing.

**Comment by biodranik** [comment]
There is an [interesting instruction](https://discuss.grapheneos.org/d/2488-organic-maps-app-voice-instructions-are-not-available/3) on how to make RHVoice work:

1. Install Organic Maps
2. Install RHVoice
3. Open RHVoice and install a language
4. Enable Talkback in Settings > Accessibility
5. Exit the tutorial and disable Talkback
6. Settings > Accessibility > Text-to-speech output
7. Make sure RHVoice is configured there
8. Open Organic Maps and the setting for voice instructions can be enabled.

**Comment by rtsisyk** [comment]
This is a documentation issue. https://github.com/organicmaps/organicmaps.github.io/pull/32

**Comment by biodranik** [comment]
No, it's not a documentation issue (or not only a documentation issue). In some cases, our old implementation of the TTS engine doesn't get changed system settings immediately and requires either a device reboot or installing several TTS engines in the system.

Unfortunately, users are still regularly reporting this issue.

Last time when I checked the TTS code, it was using the old system API and wasn't enumerating all TTS engines properly. It needs some improvements to work reliably.

**Comment by biodranik** [comment]
This issue should be resolved now. If someone still reproduces it, please let us know.
